_posts/2020-10-26-dynamically-changeable-logging.md

```diff
@@ -0,0 +1,150 @@
+---
+layout: post
+title:  "Dynamically changeable logging levels"
+date: 2020-10-26
+author: stanislav_knot
+---
+
+Logging is an indisputably important feature for applications.
+Logging monitors what the application is doing, but equally importantly,  it logs what the application is not doing. 
+Looking into application logs can help us prevent an application remaining or even getting into a faulty state.
+
+<!--more-->
+
+
+## Setting logging levels dynamically
+
+Strimzi components are split into the two groups regarding which implementation of `log4j` they use.
+
+`log4j`
+Kafka brokers
+Kafka Connect
+Kafka MirrorMaker2
+`log4j2`
+Strimzi Kafka Bridge
+Topic Operator
+User Operator
+Strimzi Cluster Operator
+
+You can change logging levels dynamically by ...
+
+Each implementation requires a different approach, which we will look at now.
+
+
+### Strimzi Cluster Operator
+
+In order to change the Cluster Operator logging config, a simple change is required.
+You have to configure the logging settings in the `ConfigMap` used for Cluster Operator configuration.
+This `ConfigMap` is by default called `strimzi-cluster-operator`,
+To edit the logging configuration use
+```
+$ kubectl edit cm strimzi-cluster-operator
+```
+
+Under the field `log4j2.properties` you should see current logging configuration.
+Now you can edit this configuration to whatever you want with one exception.
+The entry `monitorInterval` must stay in the place.
+
+The second way to change `ConfigMap` is to modify the file `install/cluster-operator/050-ConfigMap-strimzi-cluster-operator.yaml`, and apply the changes by this command:
+
+```
+$ kubectl apply -f install/cluster-operator/050-ConfigMap-strimzi-cluster-operator.yaml
+```
+
+The same restrictions apply.
+The entry `monitorInterval` must stay in the place.
+
+The change in logging takes some time to be applied. 
+You can find out more in the implementation details section of this post.
+
+### Topic Operator, User Operator and Kafka resources
+
+The approach to changing the logging configuration of User Operator, Topic Operator, Kafka Bridge, Kafka brokers, Kafka Connect and Kafka MirrorMaker2 did not change.
+The logging is still configurable via `inline` or `external` logging.
+
+Example of `inline` logging.
+```yaml
+...
+spec:
+  logging:
+    type: inline
+      loggers:
+        kafka.root.logger.level = INFO
+...
+```
+
+Example of external logging.
+```yaml
+...
+spec:
+  logging:
+    type: external
+      name: my-external-config-map
+...
+```
+
+Only loggers can be changed dynamically.
+Logging appenders are changed using an external `ConfigMap` and the rolling update is triggered. This does not apply for Kafka Bridge, User operator and Topic operator.
```

Comment:
Maybe you should spend more time on _This does not apply for Kafka Bridge, User operator and Topic operator._? It is not completely clear what does it apply to ... and how does it work there then. So I think you should add it.